custom-elements/reactions/resources/reactions.js

```diff
@@ -78,6 +78,38 @@ function testNodeDisconnector(testFunction, name) {
     container.parentNode.removeChild(container);
 }
 
+function testCloner(testFunction, name) {
+    let container = document.createElement('div');
+    container.appendChild(document.createElement('div'));
+    document.body.appendChild(container);
+
+    test(function () {
+        var element = defineNewCustomElement(['id']);
+        var instance = document.createElement(element.name);
+        container.appendChild(instance);
+
+        instance.setAttribute('id', 'foo');
+        assert_array_equals(element.log().types(), ['constructed', 'connected', 'attributeChanged']);
+        var newInstance = testFunction(instance);
+        var logEntries = element.log();
+        assert_array_equals(logEntries.types(), ['constructed', 'attributeChanged']);
+        assert_equals(logEntries.log().name, 'id');
+        assert_equals(logEntries.log().oldValue, null);
+        assert_equals(logEntries.log().newValue, 'foo');
+    }, name + ' must enqueue a attributeChanged reaction when cloning an element with an observed attribute');
+
+    test(function () {
+        var element = defineNewCustomElement(['id']);
+        var instance = document.createElement(element.name);
+        container.appendChild(instance);
+
+        instance.setAttribute('lang', 'en');
+        assert_array_equals(element.log().types(), ['constructed', 'connected']);
+        var newInstance = testFunction(instance);
+        assert_array_equals(element.log().types(), ['constructed']);
+    }, name + ' must not enqueue a attributeChanged reaction when cloning an element with an unobserved attribute');
```

Comment:
Is there a way to make this "an element with an unobserved attribute" more accurate but equally succinct? It must not queue an attribute changed callback for *unobserved attributes*; but a custom element must queue an attributeChanged callback for an observed attribute, even if there happen to be other unobserved attributes. Same goes for the earlier file.